spec/controllers/questions_controller_spec.rb

```diff
@@ -0,0 +1,130 @@
+require 'rails_helper'
+
+RSpec.describe QuestionsController, type: :controller do
+    let(:question) { create(:question) }
+
+  describe 'GET #index' do
+    let(:questions) { create_list(:question, 2) }
+    before { get :index }
+
+    it 'populates an array of all questions' do
+      expect(assigns(:questions)).to match_array(questions)
+    end
+
+    it 'renders index view' do
+      expect(response).to render_template :index
+    end
+  end
+
+  describe 'GET #show' do
+    before { get :show, id: question }
+
+    it 'assigns the requested question to @question' do
+      expect(assigns(:question)).to eq question
+    end
+
+    it 'renders show view' do
+      expect(response).to render_template :show
+    end
+  end
+
+  # метод NEW не нуждаеться в создании объекта так как сам создаёт его автоматически
+  describe 'GET #new' do
+    before { get :new}
+
+    it 'assigns a new Question to @question' do
+      expect(assigns(:question)).to be_a_new(Question)
+    end
+
+    it 'renders new view' do
+      expect(response).to render_template :new
+    end
+  end
+
+  describe 'GET #edit' do
+    before { get :edit, id: question}
+
+    it 'assigns the requested question to @question' do
+      expect(assigns(:question)).to eq question
+    end
+
+    it 'renders edit view' do
+      expect(response).to render_template :edit
+    end
+  end
+
+  describe 'POST #create' do
+    context 'with valid attributes' do
+      it 'saves new question in the DB' do
+        # old_count = Question.count
+        # post :create, question: attributes_for(:question)
+        # expect (Question.count).to eq old_count + 1
+        expect { post :create, question: attributes_for(:question) }.to change(Question, :count).by(1)
+      end
+
+      it 'redirects to show view' do
+        post :create, question: attributes_for(:question)
+        expect(response).to redirect_to question_path(assigns(:question))
+      end
+    end
+
+    context 'with invalid attributes' do
+      it 'does not saves new question in the DB' do
+        expect { post :create, question: attributes_for(:invalid_question) }.to_not change(Question, :count)
+      end
+
+      it 're-renders new view' do
+        post :create, question: attributes_for(:invalid_question)
+        expect(response).to render_template :new
+      end
+    end
+  end
+
+  describe 'PATCH #update' do
+    context 'with valid attributes' do
+      it 'assigns the requested question to @question' do
+        patch :update, id: question, question: attributes_for(:question)
+        expect(assigns(:question)).to eq question
+      end
+
+      it 'changes question attributes' do
+        patch :update, id: question, question: { title: 'New Title', body: 'New Body' }
+        question.reload
+        expect(question.title).to eq 'New Title'
+        expect(question.body).to eq 'New Body'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.